src/utils/helper.ts

```diff
@@ -45,3 +45,28 @@ export function findOrCreateRole(event: RoleGranted, grantor: Account, grantee:
   role.save()
   return role
 }
+
+export function generateRoleApprovalId(grantor: Account, operator: Account, tokenAddress: string): string {
+  return grantor.id + '-' + operator.id + '-' + tokenAddress.toLowerCase()
+}
+
+export function insertRoleApprovalIfNotExist(grantor: Account, operator: Account, tokenAddress: string): RoleApproval {
+  const roleApprovalId = generateRoleApprovalId(grantor, operator, tokenAddress)
+  let roleApproval = RoleApproval.load(roleApprovalId)
+  if (!roleApproval) {
+    roleApproval = new RoleApproval(roleApprovalId)
+    roleApproval.grantor = grantor.id
+    roleApproval.operator = operator.id
+    roleApproval.tokenAddress = tokenAddress.toLowerCase()
+    roleApproval.save()
+  }
+  return roleApproval
+}
+
+export function deleteRoleApprovalIfExist(grantor: Account, operator: Account, tokenAddress: string): void {
```

Comment:
I think this function just need to receive the entityId

```suggestion
export function deleteRoleApprovalIfExist(roleApprovalId: string): void {
```